shapely/constructive.py

```diff
@@ -993,6 +994,63 @@ def simplify(geometry, tolerance, preserve_topology=True, **kwargs):
         return lib.simplify(geometry, tolerance, **kwargs)
 
 
+@requires_geos("3.12.0")
+@multithreading_enabled
+def coverage_simplify(geometry, tolerance, simplify_boundary=True, axis=None, **kwargs):
+    """Return a simplified version of an input geometry using coverage simplification.
+
+    Assumes that the geometry forms a polygonal coverage. Under this assumption, the
+    function simplifies the edges using the Visvalingam-Whyatt algorithm, while
+    preserving a valid coverage. In the most simplified case, polygons are reduced to
+    triangles.
+
+    A collection of valid polygons is considered a coverage if the polygons are:
+
+    * **Non-overlapping** - polygons do not overlap (their interiors do not intersect)
+    * **Edge-Matched** - vertices along shared edges are identical
+
+    The function allows simplification of all edges including the outer boundaries of
+    the coverage or simplification of only the inner (shared) edges.
+
+    If there are other geometry types than Polygons or MultiPolygons present, the
+    resulting GeometryCollection will not undergo simplification and geometries are
+    returned collected but unchanged.
+
+    If the geometry is polygonal but does not form a valid coverage due to overlaps,
+    it will be simplified but it may result in invalid topology.
+
+    Parameters
+    ----------
+    geometry : Geometry or array_like
+    tolerance : float or array_like
+        The degree of simplification roughly equal to the square root of the area
+        of triangles that will be removed.
+    simplify_boundary : bool, optional
+        By default (True), simplifies both internal edges of the coverage as well
+        as its boundary. If set to False, only simplifies internal edges.
+    axis : int, optional
+        Axis along which the operation is performed. The default (None)
+        performs the operation over all axes, returning a scalar value.
+        Axis may be negative, in which case it counts from the last to the
+        first axis.
```

Comment:
`axis` is new to me. It comes from Pandas, right? I think boundaries between Pandas/GeoPandas and Shapely are a good thing and that we should resist blurring the projects if we can. What does `axis` really mean to Shapely, given that inputs can only be a scalar geometry or a 1-D array of geometries? Can't we infer axis from the type of input?

Reply:
This comes directly from `shapely.union_all` and other predicate _all functions which have the same notion of using axis to define what should be treated as a single group. So it is not new.

> What does axis really mean to Shapely, given that inputs can only be a scalar geometry or a 1-D array of geometries?

That is not correct as far as I'm aware. While some operations will not work on a N-D array (STRtree notably), most of shapely operations are perfectly capable of handling N-D arrays of geometries and there's a clear use case for that in the vector data cube space for dealing with time-dependent geometries.

If you think this should change, we should do that across all functions that use it, so I'd keep it for a separate discussion and PR.

Reply:
Yeah, since the low-level operations were written like ufuncs, I shouldn't be surprised that they would work on N-D arrays.

Even, so why would we want to ravel in one case, and rollaxis in the other? What does this mean to a GIS programmer? If `lib.coverage_simplify()` can accept N-D arrays, can the caller not ravel or rollaxis or neither (perhaps?) as they need before calling the method?

In matters of software design, I tend to be anti-convenience. Instead of accepting an `axis` argument and then modifying input based on the value, callers should just modify their inputs in their own applications. Then we have less code, fewer special cases and branches to test. I would make an exception for situations where GEOS C API methods must be called, because how is a user going to do that? But here the user already has numpy installed and can easily reshape their inputs.